native_client/client.cc

```diff
@@ -233,6 +251,59 @@ ProcessFile(ModelState* context, const char* path, bool show_times)
   }
 }
 
+std::vector<meta_word>
+WordsFromMetadata(Metadata* metadata)
+{
+  std::vector<meta_word> word_list;
+
+  std::string word = "";
+  float word_start_time = 0;
+
+  // Loop through each character
+  for (int i=0; i<metadata->num_items; i++) {
+    MetadataItem item = metadata->items[i];
+
+    unsigned char character = static_cast<unsigned char>(*item.character); 
```

Comment:
Yes, but for the purposes of checking if the character is a space, this is not a problem. Space will always be a single byte character.

Reply:
OK. Understood, the code below is
```C++
if (std::isspace(character) == 0) word.append(item.character);
```
and not
```C++
if (std::isspace(character) == 0) word.append(character);
```

Reply:
The character variable is intended solely for the isspace function, which is fussy about what it accepts. strcmp(item.character," ") would be simpler - is this likely to break any language support?

Reply:
Thinking about it more, unfortunately, this already breaks multi-language support.

Conceptually what we want is to look for a character that separates words. In English , French... this character is a space and is encoded as a single byte UTF-8 character. So this works as expected for English , French...

However for Chinese, Japanese, and Korean (CJK) there are "[ideographic spaces](https://en.wiktionary.org/wiki/ideographic_space)" which separate words and are _not_ represented as a single byte UTF-8 character, see [here](https://www.fileformat.info/info/unicode/char/3000/index.htm). Thus std::isspace() and this code will not work with such ideographic spaces.

I am not sure if this is the only case that would break this code.

Reply:
It would be easy enough to check for a list of separator characters, even if they span multiple bytes. Any thoughts @reuben, @lissyx?

Reply:
@dabinat It should be easy to test. According to [Wikipedia](https://en.wikipedia.org/wiki/Whitespace_character#Unicode) there are only 25 to check for.

Reply:
Note that breaking on ideographic spaces still won't make this code work for CJK. Ideographic spaces are only used in some cases, and the norm is for no spaces at all. Trying to make a single logic fit all languages simply will not work so I don't think we should be spending effort on that.

Reply:
Ok, in that case I'll remove the isspace and replace with strcmp but otherwise leave the logic the same.